akka-docs-dev/rst/scala/code/docs/stream/CompositionDocSpec.scala

```diff
@@ -137,17 +137,17 @@ class CompositionDocSpec extends AkkaSpec {
     //#partial-flow-dsl
     // Convert the partial graph of FlowShape to a Flow to get
     // access to the fluid DSL (for example to be able to call .filter())
-    val flow = Flow.wrap(partial)
+    val flow = Flow.fromGraph(partial)
 
     // Simple way to create a graph backed Source
-    val source = Source() { implicit builder =>
+    val source = Source.fromGraph( FlowGraph.create() { implicit builder =>
       val merge = builder.add(Merge[Int](2))
       Source.single(0)      ~> merge
       Source(List(2, 3, 4)) ~> merge
 
       // Exposing exactly one output port
-      merge.out
-    }
+      SourceShape(merge.out)
+    })
```

Comment:
If one specifies the wrong return value (not a Shape):

inferred type arguments [(akka.stream.Inlet[Int], akka.stream.Outlet[Int])] do not conform to method create's type parameter bounds [S <: akka.stream.Shape]

If one specifies the wrong shape:

[error]  found   : akka.stream.Graph[akka.stream.SinkShape[Int],akka.actor.Cancellable]
[error]  required: akka.stream.Graph[akka.stream.FlowShape[?,?],?]